c10/core/DispatchKey.h

```diff
@@ -444,6 +444,8 @@ enum class DispatchKey : uint16_t {
   Autograd,
   CompositeImplicitAutograd, // registered at
   // build/aten/src/ATen/RegisterCompositeImplicitAutograd.cpp
+  CompositeImplicitBatched, // registered at
```

Comment:
Naming bikeshed question (for Richard + others too)

What about `FuncTorchBatchedAlias`, instead of `CompositeImplicitBatched`?

The existing alias keys that we have with the word "Composite" in them include:
- CompositeImplicitAutograd
- CompositeImplicitAutogradNestedTensor
- CompositeExplicitAutograd
- CompositeExplicitAutogradNonFunctional

I think their names are already pretty confusing, but one thing that they all have in common is that they register to backend kernels in some capacity. `CompositeImplicitAutogradNestedTensor` only registers to the NestedTensor backend (because... layouts in the dispatcher are a bit weird), and the other keys register to **all** dense backends.

Compare that to this new key, which is really just an alias for `FuncTorchBatched`, but set up in such a way so that we can distinguish between a direct batching rule vs. a composite batching rule.

Reply:
I originally named it `CompositedImplicitBatched` to be consistent, but you have a good argument that it's not actually consistent.

`FuncTorchBatchedAlias` seems reasonable. Some other ideas: `FuncTorchBatchedDecomposition` to make it clearer that the key is for decompositions

Reply:
I think `FuncTorchBatchedDecomposition` is better than `FuncTorchBatchedAlias` (as mentioned above for clarity)

Reply:
I've updated the name to `FuncTorchBatchedDecomposition`